src/pages/OnboardingPrivateDomain/BaseOnboardingPrivateDomain.tsx

```diff
@@ -48,6 +50,21 @@ function BaseOnboardingPrivateDomain({shouldUseNativeStyles, route}: BaseOnboard
         resendValidateCode(email);
     }, [email]);
 
+    const handleBackButtonPress = useCallback(() => {
```

Comment:
```suggestion
    const handleBackButtonPress = useCallback(() => {
        if (onboardingValues?.shouldValidate === false) {
            updateOnboardingValuesAndNavigation(onboardingValues);
            return;
        }
        const routeToNavigate = (route.params?.backTo as Route) ?? ROUTES.ONBOARDING_PERSONAL_DETAILS.getRoute();
        Navigation.goBack(routeToNavigate);
    }, [route.params?.backTo, onboardingValues]);
```

I think we should use the shouldValidate flag to indicate the case of adding a work email. Please help update it and remove any unused imports